packages/core/src/controllers/OptionsController.ts

```diff
@@ -130,6 +130,11 @@ export interface OptionsControllerStatePublic {
    * @default false
    */
   enableEmbedded?: boolean
+  /**
+   * Enable manual control of WalletConnect connections.
+   * @default false
+   */
+  enableUniversalProviderManualControl?: boolean
```

Comment:
Do we need this flag to expose to the users? If I'm not missing another detail, we could keep it as an internal state depending on the given `walletConnectProvider`, if they provide it this will be `true`, otherwise it'll be `false` by default. Just to keep the extra props as low as possible

Reply:
Agree with Enes here. I think this can confuse user's.

Reply:
We can, the flag was to provide an option for delegating control of the provider entirely